op-node/rollup/derive/batches.go

```diff
@@ -224,28 +250,36 @@ func checkSpanBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1BlockRe
 		return BatchUndecided
 	}
 
-	if startEpochNum < epoch.Number {
-		log.Warn("dropped batch, epoch is too old", "minimum", epoch.ID())
+	if startEpochNum < parentBlock.L1Origin.Number {
+		log.Warn("dropped batch, epoch is too old", "minimum", parentBlock.ID())
 		return BatchDrop
 	}
 
 	originIdx := 0
 	originAdvanced := false
-	if startEpochNum == epoch.Number+1 {
+	if startEpochNum == parentBlock.L1Origin.Number+1 {
 		originAdvanced = true
 	}
 
 	for i := 0; i < batch.GetBlockCount(); i++ {
+		if batch.GetBlockTimestamp(i) <= l2SafeHead.Time {
+			continue
+		}
+		var l1Origin eth.L1BlockRef
+		for j := originIdx; j < len(l1Blocks); j++ {
```

Comment:
Why do we need this loop? Also, what happens if l1Origin is not found and uninitialized?

Reply:
When we check OriginCheck, It's already confirmed that we have enough L1blocks to cover all span batch.
The reason why I have changed this loop is the existing code does not work when `l1Blocks[0] != safeHead.origin`

Reply:
If we check originCheck, we can say that l1 block corresponding to `endEpochNum` is known.
> , It's already confirmed that we have enough L1blocks to cover all span batch.
Can you elaborate?

Reply:
we don't need l1block corresponding to startEpoch. we need l1block for descendants of l2safehead